pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -75,6 +77,8 @@ public ResultTable submitAndReduce(long requestId, QueryPlan queryPlan,
   public int submit(long requestId, QueryPlan queryPlan, long timeoutMs)
       throws Exception {
     int reduceStageId = -1;
+    long submissionDeadlineMs = System.currentTimeMillis() + DEFAULT_SUBMISSION_DEADLINE_MS;
```

Comment:
this timeout is only until the servers receive the query, right? it has nothing to do with executing the query? if so, 1s seems like a pretty large default (but probably OK)

Reply:
yeah i am not so convince this is the right way to do async dispatch. i am open to all suggestions. please definitely share your feedback

Reply:
@61yao @siddharthteotia @ankitsultana any idea on what would be the best solution?